app/code/core/Mage/Api2/Model/Resource.php

```diff
@@ -1030,7 +1030,7 @@ public function getEavAttributes($onlyVisible = false, $excludeSystem = false)
         $model = $this->getConfig()->getResourceWorkingModel($this->getResourceType());
 
         /** @var Mage_Eav_Model_Entity_Type $entityType */
-        $entityType = Mage::getModel('eav/entity_type')->load($model, 'entity_model');
+        $entityType = Mage::getSingleton('eav/config')->getEntityType($model, 'entity_model');
```

Comment:
`$model` is one of 
![image](https://user-images.githubusercontent.com/1106470/216209534-40833be9-3911-430a-82e5-a1d6b5d5306c.png)
When I try
```php
$entityType = Mage::getSingleton('eav/config')->getEntityType('customer/customer', 'entity_model');
```
I get
> Invalid entity_type specified: customer/customer

I suggest reverting this change. The original load takes about 0.7ms anyway in my development server.

Reply:
I've just tested this and it works perfectly fine. Reverting is not an option, the whole idea is to reduce the repeated and unnecessary loading of EAV-type entities. I need more information to reproduce this.
Are you using the latest version of the PR (there was a change half-way that might affect this)?
Which context did you test this in? Is it a unmodified Openmage install?

Reply:
I am testing this function specifically to see if it can load by `entity_model`: https://github.com/OpenMage/magento-lts/blob/3d3e9163461d9fa2736c7340a68f35efbd9d10a6/app/code/core/Mage/Eav/Model/Config.php#L295-L295

The simple test says it cannot.  I test it by simply running this single statement:
```php
$entityType = Mage::getSingleton('eav/config')->getEntityType('customer/customer', 'entity_model');
```
I replaced `$model` with string `'customer/customer'`.

Note that the original load
```php
$entityType = Mage::getModel('eav/entity_type')->load($model, 'entity_model');
```
is load by `entity_model`. I think you need to change `$model` to `entity_type_code` in order to use `Mage::getSingleton('eav/config')->getEntityType()`.

Reply:
Kiatng: The code reference you linked is old, I extended the `Mage_Eav_Model_Config::getEntityType($code)` with a second parameter `$field` that mimics the abstract models `load(...)` second parameter.

Reply:
Can you provide the extended code for review? I cannot find it here.

Reply:
It's in the PR, that code snipped was linked directly from the PR changes.

Reply:
Thanks, I found it here: https://github.com/OpenMage/magento-lts/blob/fa31e6067d41ec80a725650d8f2fcad2b732eacf/app/code/core/Mage/Eav/Model/Config.php